core/parser.mly

```diff
@@ -883,7 +883,7 @@ database_expression:
 | INSERT exp VALUES LPAREN record_labels RPAREN exp            { db_insert ~ppos:$loc $2 $5 $7 None }
 | INSERT exp VALUES LBRACKET LPAREN loption(labeled_exps)
   RPAREN RBRACKET preceded(RETURNING, VARIABLE)?               { db_insert ~ppos:$loc $2 (labels $6) (db_exps ~ppos:$loc($6) $6) $9  }
-| INSERT exp VALUES LPAREN record_labels RPAREN db_expression
+| INSERT exp VALUES LPAREN record_labels RPAREN exp
```

Comment:
This fixes part of #563.

Reply:
I'm now getting the following warning:
```
      menhir core/parser.{ml,mli}
Warning: 2 states have shift/reduce conflicts.
Warning: 2 shift/reduce conflicts were arbitrarily resolved
```

Reply:
Yes, I'm not sure why that is.  What was there before was definitely wrong (`db_expression` only derives delete/update expressions).

Reply:
The reason is that `db_expression` used to have a fallthrough case to `typed_expression`, this was to avoid shift-reduce conflicts.  At some point, the fallthrough case was removed.  The right thing to do is probably to allow these INSERTs to end with `typed_expression` since update and delete operations in this position will never typecheck anyway.

Reply:
Sadly, Menhir is always right when it reports conflicts. I was about to explain why the conflicts were happening but I'm glad I no longer have to :-)

Also, be careful with trailing whitespaces. I wonder whether CI will catch them?

Reply:
>  I wonder whether CI will catch them?

Yes, the rule checking script rejects certain files with trailing whitespace, including the parser files.

Reply:
Indeed:

```
$ make rule-check
Applying rule check
Running 'tools/rules/trailing_whitespace'...
  - core/parser.mly
make: *** [rule-check] Error 1
```

Reply:
Correct, my emacs for some reason inserts trailing whitespace for no apparent reason.